### PR TITLE
GH-15556 Implement UpliftDRF MOJO

### LINKED ISSUE #15556: Add MOJO to UpliftDRF model
Add Mojo support to all uplift models.  This task can be broken into sub-tasks if deem necessary.

### PR DESCRIPTION
**Pull request by maurever**
Closes #15556.

### CODE REVIEW

File: h2o-core/src/main/java/hex/ModelMetricsBinomialUplift.java [hunk 1]
```diff
@@ -58,32 +59,64 @@ protected StringBuilder appendToStringMetrics(StringBuilder sb) {
...
-     * @param targetClassProbs A Vec containing target class probabilities
+     * @param predictedProbs A Vec containing predicted probabilities
+     * @param actualLabels A Vec containing the actual labels (can be for fewer labels than what's in domain, since the predictions can be for a small subset of the data)
+     * @param treatment A Vec containing the treatment values               
+     * @param auucType Type of default AUUC
+     * @param nbins Number of bins to calculate AUUC (-1 means default value 1000, the number has to be higher than zero)  
+     * @param customAuucThresholds custom threshold to calculate AUUC, if is not specified, the thresholds will be calculated from prediction vector             
+     * @return ModelMetrics object
+     */
+    static public ModelMetricsBinomialUplift make(Vec predictedProbs, Vec actualLabels, Vec treatment, AUUC.AUUCType auucType, int nbins, double[] customAuucThresholds) {
```
Comment (maurever) [inline h2o-core/src/main/java/hex/ModelMetricsBinomialUplift.java:83]: This part could be in a different PR, but it is necessary for tests. But I can move it.

File: h2o-core/src/main/java/water/api/ModelMetricsHandler.java [hunk 2]
```diff
@@ -372,6 +378,10 @@ public static final class ModelMetricsMakerSchemaV3 extends SchemaV3<ModelMetric
             level = API.Level.secondary, direction = API.Direction.INOUT, gridable = true)
     public int auuc_nbins;
 
+    @API(help = "Custom AUUC thresholds (for uplift binomial classification).",
```
Comment (maurever) [inline h2o-core/src/main/java/water/api/ModelMetricsHandler.java:381]: These changes enable custom thresholds to calculate metrics and could be in a different PR, but it is necessary for tests. But I can move it.

File: h2o-py/h2o/h2o.py [hunk 3]
```diff
@@ -1969,7 +1969,7 @@ def load_dataset(relative_path):
 
 
 def make_metrics(predicted, actual, domain=None, distribution=None, weights=None, treatment=None, auc_type="NONE",
-                 auuc_type="AUTO", auuc_nbins=-1):
+                 auuc_type="AUTO", auuc_nbins=-1, custom_auuc_thresholds=None):
```
Comment (maurever) [inline h2o-py/h2o/h2o.py:1972]: These changes enable custom thresholds to calculate metrics and could be in a different PR, but it is necessary for tests. But I can move it.

File: h2o-py/h2o/model/model_base.py [hunk 4]
```diff
@@ -511,11 +511,16 @@ def model_performance(self, test_data=None, train=False, valid=False, xval=False
                               data={"auc_type": auc_type})
             elif auuc_type is not None:
                 assert_is_type(auuc_type, Enum("AUTO", "qini", "gain", "lift"))
+                assert_is_type(custom_auuc_thresholds, [float], None)
```
Comment (maurever) [inline h2o-py/h2o/model/model_base.py:514]: These changes enable custom thresholds to calculate metrics and could be in a different PR, but it is necessary for tests. But I can move it.

File: h2o-r/h2o-package/R/models.R [hunk 5]
```diff
@@ -1039,7 +1039,7 @@ h2o.feature_frequencies <- feature_frequencies.H2OModel
 #' h2o.performance(model = prostate_gbm_balanced, train = TRUE)
 #' }
 #' @export
-h2o.performance <- function(model, newdata=NULL, train=FALSE, valid=FALSE, xval=FALSE, data=NULL, auc_type="NONE") {
+h2o.performance <- function(model, newdata=NULL, train=FALSE, valid=FALSE, xval=FALSE, data=NULL, auc_type="NONE", custom_auuc_thresholds=NULL) {
```
Comment (maurever) [inline h2o-r/h2o-package/R/models.R:1042]: These changes enable custom thresholds to calculate metrics and could be in a different PR, but it is necessary for tests. But I can move it.

File: h2o-py/h2o/model/model_base.py [hunk 6]
```diff
@@ -460,16 +460,15 @@ def training_model_metrics(self):
         return self._model_json["output"]["training_metrics"]._metric_json
     
     def model_performance(self, test_data=None, train=False, valid=False, xval=False, auc_type=None, 
-                          auuc_type=None, auuc_nbins=-1):
```
Comment (maurever) [inline h2o-py/h2o/model/model_base.py:463]: This parameter is not able to be used. Default model thresholds are used to calculate AUUC. Change in this function is not allowed.

**Review by wendycwong — APPROVED** [review]

**Pull request merged**